hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
I would say that `movb{r, r}` is a trivial nop by definition (also, note that this is only possible when both src and dst regs are physical).  The destination is being interpreted as a Reg8, which means its upper bits are logically garbage after the `movb{}`.  Anywhere that you require the zeroing behavior you should do an explicit `movzbq{}` (or similar).